clinical-domain-agent/src/test/java/care/smith/fts/cda/impl/TCACohortSelectorIT.java

```diff
@@ -0,0 +1,211 @@
+import static com.github.tomakehurst.wiremock.client.WireMock.ok;
+import static com.github.tomakehurst.wiremock.client.WireMock.post;
+import static org.springframework.http.HttpHeaders.CONTENT_TYPE;
+import static org.springframework.http.HttpStatus.BAD_REQUEST;
+import static org.springframework.util.MimeTypeUtils.APPLICATION_JSON_VALUE;
+import static reactor.test.StepVerifier.create;
+
+import care.smith.fts.api.ConsentedPatient;
+import care.smith.fts.cda.impl.mock.MockCohortSelector;
+import care.smith.fts.test.connection_scenario.AbstractConnectionScenarioIT;
+import care.smith.fts.util.HttpClientConfig;
+import care.smith.fts.util.WebClientFactory;
+import care.smith.fts.util.error.TransferProcessException;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.github.tomakehurst.wiremock.client.MappingBuilder;
+import com.github.tomakehurst.wiremock.client.WireMock;
+import com.github.tomakehurst.wiremock.junit5.WireMockRuntimeInfo;
+import com.github.tomakehurst.wiremock.junit5.WireMockTest;
+import io.micrometer.core.instrument.MeterRegistry;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Stream;
+import lombok.extern.slf4j.Slf4j;
+import org.hl7.fhir.r4.model.*;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.ProblemDetail;
+import reactor.core.publisher.Flux;
+
+@Slf4j
+@SpringBootTest
+@WireMockTest
+class TCACohortSelectorIT extends AbstractConnectionScenarioIT {
+
+  @Autowired MeterRegistry meterRegistry;
+  @Autowired ObjectMapper om;
+  private WireMock wireMock;
+
+  private static final Set<String> POLICIES = Set.of("MDAT_erheben");
+
+  private static final String PID_SYSTEM =
+      "https://ths-greifswald.de/fhir/gics/identifiers/Pseudonym";
+  private static final String POLICY_SYSTEM =
+      "https://ths-greifswald.de/fhir/CodeSystem/gics/Policy";
+
+  private static TCACohortSelector cohortSelector;
+
+  private static MockCohortSelector allCohortSelector;
+  private static MockCohortSelector listCohortSelector;
+
+  @BeforeEach
+  void setUp(WireMockRuntimeInfo wireMockRuntime, @Autowired WebClientFactory clientFactory) {
+    var address = "http://localhost";
+    var server = new HttpClientConfig(address);
+    var config = new TCACohortSelectorConfig(server, PID_SYSTEM, POLICY_SYSTEM, POLICIES, "MII");
+    cohortSelector =
+        new TCACohortSelector(
+            config, clientFactory.create(clientConfig(wireMockRuntime)), meterRegistry);
+    wireMock = wireMockRuntime.getWireMock();
+    allCohortSelector = MockCohortSelector.fetchAll(wireMock, POLICY_SYSTEM);
+    listCohortSelector = MockCohortSelector.fetch(wireMock, POLICY_SYSTEM);
+  }
+
+  private static MappingBuilder getBuilderFetchAll() {
```

Comment:
getBuilder() prefix is unnecessary, as it doesn't help with understanding what the method does. It already returns a mappingBuilder, it should be called something like fetchAllRequest(). Also each getBuilder() method is only used once, could be inlined.